src/scene/shader-lib/program-library.js

```diff
@@ -151,6 +151,12 @@ class ProgramLibrary {
             const generatedShaderDef = this.generateShaderDefinition(generator, name, generationKey, options);
             Debug.assert(generatedShaderDef);
 
+            if (processingOptions.userAttributes && processingOptions.userAttributes.size > 0) {
```

Comment:
if standard material has the attributes, the standard shader generator should handle this, and not a program-library.